src/db/model/user.rs

```diff
@@ -1,30 +1,29 @@
 use crate::db::schema::users;
-use actix_web::{body::BoxBody, http::header::ContentType, HttpRequest, HttpResponse, Responder};
 use diesel::prelude::*;
 use serde::{Deserialize, Serialize};
-use serde_json;
 
 #[derive(Identifiable, Queryable, Serialize, Deserialize)]
 pub struct User {
   pub id: u64,
   pub role: String,
   pub email: String,
   pub password: String,
+  pub status: String,
 }
 
-#[derive(Insertable, Serialize, Deserialize, AsChangeset)]
+#[derive(Insertable, Serialize, Deserialize)]
 #[diesel(table_name = users)]
 pub struct NewUser {
   pub role: String,
   pub email: String,
   pub password: String,
 }
 
-impl Responder for User {
-  type Body = BoxBody;
-  fn respond_to(self, _: &HttpRequest) -> HttpResponse<Self::Body> {
-    HttpResponse::Ok()
-      .content_type(ContentType::json())
-      .body(serde_json::to_string(&self).unwrap())
-  }
+#[derive(Serialize, Deserialize, AsChangeset)]
+#[diesel(table_name = users)]
+pub struct UpdateUser {
+  pub role: Option<String>,
+  pub email: Option<String>,
+  pub status: Option<String>,
+  pub password: Option<String>,
```

Comment:
If this is something that the user can do to their own profile, an "old password" field would be required, and verified in the service. I recommend we do it.

Reply:
The current files in the application are intended to be used by admins. 
In the future, a separate endpoint for employees will be added